build/templates/_grpc_stub_interpreter.py.mako

```diff
@@ -43,9 +43,12 @@ class GrpcStubInterpreter(object):
     def get_session_handle(self):
         return self._${config['session_handle_parameter_name']}
 
-    def _invoke(self, func, request):
+    def _invoke(self, func, request, metadata=None):
         try:
-            response = func(request)
+            if metadata is not None:
+                response = func(request, metadata=metadata)
+            else:
+                response = func(request)
```

Comment:
Why do we need the if/else? Does func not already have a metadata argument with a default of None?

Reply:
I see there's already another thread about this.

Reply:
Yep. I'll copy the answer here:
For python itself, it's not necessary, but passing along metadata=None would require us to update all the assert_called_with invocations in the tests.

Reply:
> For python itself, it's not necessary, but passing along metadata=None would require us to update all the assert_called_with invocations in the tests.

Can you do this in its own follow up PR?

Reply:
Did it here. Turns out it wasn't as bad as I was imagining.